### PR TITLE
Windows xinclude patch

### PR DESCRIPTION
**Pull request by ollietb**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

XIncludes throw the following error on Windows

```
An exception has been thrown during the rendering of a template ("[WARNING 1549] failed to load external entity "xinclude/_experiences_section.xml" (in n/a - line 0, column 0)
[ERROR 1604] could not load xinclude/_experiences_section.xml, and no fallback was found (in C:\xampp2\htdocs\cms\app/Resources/pages\social_wall.xml - line 286, column 0)
[ERROR 1871] Element '{http://www.w3.org/2001/XInclude}include': This element is not expected. (in C:\xampp2\htdocs\cms\app/Resources/pages\social_wall.xml - line 286, column 0)").
```

#### Example Usage

~~~xml
<xi:include href='xinclude/_testimonials_section.xml'/>
~~~

### CODE REVIEW

**Comment by danrot** [comment]
Are the `@` necessary? @gisostallenberg Have you thought about that solution back then?

**Comment by danrot** [comment]
@ollietb You should also remove the if here: https://github.com/sulu/sulu/blob/develop/src/Sulu/Component/Content/Tests/Unit/Metadata/Loader/XmlLegacyLoaderTest.php#L1307

And an entry in the `CHANGELOG.md` file is missing 😃

**Comment by ollietb** [comment]
@danrot I copied the fix from Drupal and they have the @ so I thought it was probably there for a reason
https://api.drupal.org/api/drupal/vendor!phpunit!phpunit!src!Util!XML.php/function/PHPUnit_Util_XML%3A%3Aload/8.0.x

**Comment by ollietb** [comment]
@danrot I can't get the travis tests passing

**Comment by danrot** [comment]
@ollietb That's not your fault... We are experiencing some problems with jackrabbit in that regard. Looks like we are shooting too much data at it within a very short time, and it can't always handle it...

File: src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php [hunk 1]
```diff
@@ -106,6 +109,8 @@ function (\DOMDocument $dom) use ($resource, $schemaPath) {
             }
         );
 
+        @chdir($cwd);
```
Comment (danrot) [inline src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php]: Can you try it without the two `@`? They might just hide meaningful errors...

File: src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php [hunk 2]
```diff
@@ -95,6 +95,10 @@ public function load($resource, $type = 'page')
 
         $schemaPath = __DIR__ . static::SCHEME_PATH;
 
+        $cwd = getcwd();
+        // mute errors for non-Windows machines that have chdir disabled by their hosting provider
```
Comment (danrot) [inline src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php]: Maybe also mention that this is only totally necessary for windows machines, and linux/mac would still work without successfully executing that command. And keep the 120 char limit 😉

**Comment by danrot** [comment]
Merged in 3494ec31427ee5a2a26e10258e63337770f14ac2, because there was a conflict in the `CHANGELOG`. Thank you for the contribution! 😃

**Pull request closed without merging**